storage/storage-interface/src/lib.rs

```diff
@@ -519,50 +514,28 @@ pub trait DbReader: Send + Sync {
 }
 
 impl MoveStorage for &dyn DbReader {
-    fn batch_fetch_resources(&self, access_paths: Vec<AccessPath>) -> Result<Vec<Vec<u8>>> {
-        self.batch_fetch_resources_by_version(access_paths, self.fetch_synced_version()?)
+    fn fetch_resource(&self, access_path: AccessPath) -> Result<Vec<u8>> {
+        self.fetch_resource_by_version(access_path, self.fetch_synced_version()?)
     }
 
-    fn batch_fetch_resources_by_version(
+    fn fetch_resource_by_version(
         &self,
-        access_paths: Vec<AccessPath>,
+        access_path: AccessPath,
         version: Version,
-    ) -> Result<Vec<Vec<u8>>> {
-        let addresses: Vec<AccountAddress> = access_paths
-            .iter()
-            .collect::<HashSet<_>>()
-            .iter()
-            .map(|path| path.address)
-            .collect();
-
-        let results = addresses
-            .iter()
-            .map(|addr| self.get_account_state_with_proof_by_version(*addr, version))
-            .collect::<Result<Vec<_>>>()?;
-
-        // Account address --> AccountState
-        let account_states = addresses
-            .iter()
-            .zip_eq(results)
-            .map(|(addr, (blob, _proof))| {
-                let account_state = AccountState::try_from(&blob.ok_or_else(|| {
+    ) -> Result<Vec<u8>> {
+        let account_state = AccountState::try_from(
+            &self
+                .get_account_state_with_proof_by_version(access_path.address, version)?
+                .0
```

Comment:
FYI: I generally try to avoid using the tuple index annotation (e.g., `x.0`) because I think it makes it hard for the reader to know what data is being accessed. For example, here, I had to look up the return type of `get_account_state_with_proof_by_version` to see what data we were grabbing. But, others seem to use it, so this isn't a blocker. Just something to call out 😄

Reply:
Re-wrote the code to avoid using tuple index.
```
let (account_state_blob, _) =
            self.get_account_state_with_proof_by_version(access_path.address, version)?;
let account_state =
            AccountState::try_from(&account_state_blob.ok_or_else(|| {
                format_err!("missing blob in account state/account does not exist")
            })?)?;
```